### ISSUE TITLE
Double Fear Ward

### ISSUE THREAD

**Issue by Peonstomperx** [ccshiro/cc-buglist#1479]
Casting Fear Ward during the spell delay of fears should immune the fear and keep the fear ward.

http://www.ownedcore.com/forums/world-of-warcraft/world-of-warcraft-exploits/174691-priest-double-fear-ward.html

**Comment by SuddenDeath** [comment]
This seemed to be an exploit.

Kind of like seal twisting.

**Comment by Peonstomperx** [comment]
@SuddenDeath it's not really broken like seal twisting, just a nice trick that rewards skillful play

**Comment by Chulain** [comment]
when was this fixed on retail?

**Comment by Doppelmoral** [comment]
Nice source though. 1 random guy on a hacker forum posting such crap few days before 3.0.2 (WotLK pre-patch) and you report this abuse as intended mechanic for TBC here ?

**Comment by Peonstomperx** [comment]
I don't think it was ever fixed, probably they just kept it in there cus it's a cool trick.

http://www.mmo-champion.com/threads/1121323-Is-there-a-quot-point-quot-to-fear-ward/page3
 2012-04-25: "My point was with regard to the earlier mention of fearwarding yourself to stop the fear and keep fear ward on instant fears, and lucettes claim that this was "pretty basic" - warding a lock is like deathing a sheep, i'll accept that as "basic."

u pist locks?

**Comment by SuddenDeath** [comment]
Well, not sure how this would be implemented without knowing a lot more about the spell batch mechanics that retail had. Only way would probably be some dirty hack "fix". 

Since when you cast buffs on yourself, they are always cast instantly, there's no batch system for spells you cast on yourself. While incoming fears will follow the batch system, how would one implement it?


Still was clearly a bug, I don't see much use in it either, the priest would have to be saving his fear ward and be careful with his GCDs before a lock casts fear on him to even be able to do it. You'd then also be having full time left on the CD instead of a shorter time if you were to buff yourself with it earlier. Thinking mostly about arena cases here.

**Comment by Peonstomperx** [comment]
this is less of a bug than petattacking stealth and breaking vanishes(even with fucking cloak up), since they were fixed in later expansions

**Comment by rezzybaws** [comment]
Ahh this old trick. The fix is that the Fear Ward buff should only be removed upon being the (potential) target of a SUCCESSFUL FEAR CAST, not upon the Fear effect itself.

**Comment by Peonstomperx** [comment]
@rezzybaws I'm not so sure about that 
![fearward](https://cloud.githubusercontent.com/assets/12783794/9569419/491b6aa8-4f6b-11e5-86d0-5182396881b9.jpg)

**Comment by Jumpka** [comment]
Just for some more proof:
https://www.youtube.com/watch?v=KoQDA9mRDts#t=16m51s

**Comment by ccshiro** [comment]
This would require **a lot** of research to understand why it happens, and how to properly implement it.

**Comment by Jumpka** [comment]
It actually looks 100% the same as when you tremor a fear here on CC at the very end, where you get instant pulse thanks to spell delay. You don't lose control of your character (can't make video of the CC version with tremor, but it looks exactly like in https://www.youtube.com/watch?v=KoQDA9mRDts#t=16m51s)
Couldn't there be some workaround where fear ward would behave exactly like tremor totem but simly give the fear ward buff right after the "pulse" ?

**Comment by Jumpka** [comment]
Just a note since everyone thinks it's much less important than it actually is, this is definitelly not some one in a million trick, but something priests on AT try to do most of the time with pretty high sucess rate imo. (AT implemented it with some workaround, but you can see priests doing it in lot of priest videos)
Also to Sd, it's generally used against a priest fear, not a warlock

**Comment by Schaka** [comment]
I used to do this fairly frequently on retail too. It's definitely NOT an unimportant mechanic that should be disregarded. The way I understand it works is fairly simple:
If you cast FW between the time fear was put on the next batch and the time the batch is executed (fear applied to you), it will cause the desired effect.

**Comment by Peonstomperx** [comment]
the immune message is wrong too atm, it shouldn't be delayed until the batch (it shouldn't show immune either if u "preshot" it), it should be like this:
```
fearsuccess
if fearward
	msg_immune
	removeward = true
else 
	fear = true

spellbatch
if fear and not fearward
	apply fear
elseif removeward
	remove ward
```
also this isn't low prio, stop being such lock sympathizers
